### DISCUSSION TITLE
Multiple (more than 2 agents) self_chat

### DISCUSSION THREAD

**Discussion by MagicDinosaur** [facebookresearch/ParlAI#4220] [Q&A]
Hi!
I am entirely new to parlai. I wanted to know is that impossible for me to generate a self-chat model (similar to this one), but with three or more "agents" instead of only 2. So far, i have added one more agent into "parlai/scripts/self_chat.py" which is based on the original code, changed the input world of SelfChatWorld(DialogPartnerWorld) (line 67 - parlai/tasks/self_chat/worlds.py) to SelfChatWorld(MultiAgentDialogWorld). But I still got the dialog that contained only two agents.
I am wondering if I could get some expert advice on what steps needed to be done. Am I on the right track or not? Is there anything I need to edit? I would like to thank you very much for reading my question, and I hope I can get some help from you guys.

**Comment by stephenroller** [comment]
You definitely can, but you'll need to manually manage observe/self_observe/act yourself.
